### PR TITLE
Make Server Configs Editable

### PR DESCRIPTION
**Pull request by zulqarnainhanif**
## Summary

Made the below fields editable:

Server:
* gaTrackingId

Client:
* logo
* title

Email:
* everything other than smsNameCharacterLimit

Authentication:
* oauth.facebook.key
* oauth.facebook.secret
* oauth.github.key
* oauth.github.secret
* oauth.google.key
* oauth.google.secret
* oauth.linkedin.key
* oauth.linkedin.secret
* oauth.twitter.key
* oauth.twitter.secret

AWS:
* everything in sms

using below data from database instead of publicRuntimeConfig in client workspace
* title (same as config.client.title; also, siteTitle should be removed and the places where siteTitle is used should be converted to title)
* faviconXXpx/iconXXXpx
* siteDescription
* logo
* auth.<everything> - these should be sourced from authentication.authStrategies

Added and used below configs

* smsMagicLink in db configs
* emailMagicLink in db configs

Updated the profile menu to reflect auth strategy enabled changes.
Updated different layouts meta link tags though helmet sourcing data through configs

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [x] Changes have been manually QA'd
- [x] Changes reviewed by at least 2 approved reviewers

## References

https://github.com/XRFoundation/XREngine/issues/2704

## Reviewers

@HexaField @barankyle

### CODE REVIEW

**Comment by barankyle** [comment]
Looks like a good start. When I changed one of the auth keys and hit save, the JSON saved to the authentication.oauth column was mostly blank, but I'm sure that's a simple fix.

**Comment by zulqarnainhanif** [comment]
> Looks like a good start. When I changed one of the auth keys and hit save, the JSON saved to the authentication.oauth column was mostly blank, but I'm sure that's a simple fix.

yeap, server config, client config, authentication config are done and working fine now. moving on to aws and email.

File: packages/client-core/src/admin/components/Setting/Authentication.tsx [hunk 1]
```diff
@@ -71,7 +86,7 @@ const Account = (props: Props) => {
     if (user?.id?.value != null && authSettingState.updateNeeded.value) {
       AuthSettingService.fetchAuthSetting()
     }
-  }, [authState.user?.id?.value])
+  }, [authState])
```
Comment (speigg) [inline packages/client-core/src/admin/components/Setting/Authentication.tsx:89]: This won't work unfortunately. Hookstate state objects will change on every component render, so you have to use the actual values for deps here.

Comment (speigg) [reply packages/client-core/src/admin/components/Setting/Authentication.tsx:89]: Eventually (hopefully soon), we'll be able to pass the state the way you did; this is dependent on new feature we are sponsoring in Hookstate.

Comment (zulqarnainhanif) [reply packages/client-core/src/admin/components/Setting/Authentication.tsx:89]: will revert this change and copy the same for other files as well

**Review by HexaField — APPROVED** [review]

**Review by mrhegemon — APPROVED** [review]

**Review by barankyle — APPROVED** [review]

**Pull request merged**